### DISCUSSION TITLE
(401) Unauthorized Error

### DISCUSSION THREAD

**Discussion by DrMarkX** [Snow-Shell/servicenow-powershell#210] [Q&A] [answered]
Is anyone else experiencing issues when getting data you are rejected with an unauthorized error?  This happens if the get is for several tasks at once and once the error is encountered you cannot query ServiceNow for ~20 minutes.  We've opened a ticket with ServiceNow to investigate but wanted to see if this community is seeing this issue too.

**Error message**
```
Invoke-WebRequest : The remote server returned an error: (401) Unauthorized.
At C:\Program Files\WindowsPowerShell\Modules\ServiceNow\3.4.2\Private\Invoke-ServiceNowRestMethod.ps1:144 char:21
+         $response = Invoke-WebRequest @params
+                     ~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-WebRequest], WebException
+ FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeWebRequestCommand
```

**Comment by gdbarron** [comment, marked as answer]
If you want to provide some sample code I can try and see if I get the same result.

**Reply by DrMarkX** [reply]
`$tasks = @(Get-ServiceNowRecord -Table 'catalog task' -First 20 -Property number,short_description,sys_created_on,sys_updated_on,request_item.number,sys_id)`
This is a very short version of one of our scripts that fail.  This get causes the unauthorized error to occur.  It is random.  You may be able to run it once without issue, then rerun it and get the error and then you get the error immediately if you rerun it.

**Reply by DrMarkX** [reply]
I think we have figured it out. It's not a problem with the module or ServiceNow. We found our non-prod environments has a bad LDAP password that would lock the account causing anyone running our scripts to get the error. That was totally random and hard to figure out from the SNOW logs. Thanks for trying to assist.

**Reply by pmt205** [reply]
> I think we have figured it out. It's not a problem with the module or ServiceNow. We found our non-prod environments has a bad LDAP password that would lock the account causing anyone running our scripts to get the error. That was totally random and hard to figure out from the SNOW logs. Thanks for trying to assist.

Hi, I'm currently experiencing the same issue, have you found any way to fix it?

**Reply by DrMarkX** [reply]
We created new accounts for our non-prod and prod environments then had the SN engineers enter the new credentials.

**Reply by pmt205** [reply]
> We created new accounts for our non-prod and prod environments then had the SN engineers enter the new credentials.

If I'm just a normal user, which means I can't creat account or do anything to the DB, I just want to create an app to access to ServiceNow with Powershell, is there anything I can do from my side without help from SNOW team?

**Reply by gdbarron** [reply]
Unfortunately not, you need api access and they need to grant it.

**Reply by pmt205** [reply]
> Unfortunately not, you need api access and they need to grant it. 

Oh okay, thank you guys very much
